src/Core/Workspace/Project.cs

```diff
@@ -184,13 +201,66 @@ public class Project
         /// <param name="rootFolder">The path to the workspace folder.</param>
         /// <param name="cacheFolder">The path to the assembly cache folder.</param>
         /// <param name="skipAutoLoadProject">Whether to skip auto-loading the project file.</param>
+        /// <param name="logger">Used for reporting log messages.</param>
         /// <returns>The created <see cref="Project"/> object.</returns>
-        internal static Project FromWorkspaceFolder(string rootFolder, string cacheFolder, bool skipAutoLoadProject)
+        internal static Project FromWorkspaceFolder(string rootFolder, string cacheFolder, bool skipAutoLoadProject, ILogger? logger = null)
         {
+            var defaultProject = new Project(string.Empty, rootFolder, cacheFolder);
+            if (skipAutoLoadProject)
+            {
+                logger?.LogInformation("Not looking for .csproj to load automatically, because skipAutoLoadProject was specified " +
+                    "via --skipAutoLoadProject command line switch or via IQSHARP_SKIP_AUTO_LOAD_PROJECT environment variable.");
+                return defaultProject;
+            }
+
             var projectFiles = Directory.EnumerateFiles(rootFolder, "*.csproj", SearchOption.TopDirectoryOnly);
-            return (skipAutoLoadProject || projectFiles.Count() != 1)
-                ? new Project(string.Empty, rootFolder, cacheFolder)
-                : Project.FromProjectFile(projectFiles.Single(), cacheFolder);
+            if (!projectFiles.Any())
+            {
+                logger?.LogInformation($"No .csproj file found in workspace root {rootFolder}. Will compile only .qs files in this folder.");
+                return defaultProject;
+            }
+
+            var projects = projectFiles
+                .Select(projectFile => Project.FromProjectFile(projectFile, cacheFolder))
+                .Where(project => project.UsesQuantumSdk);
+            if (!projects.Any())
+            {
+                logger?.LogInformation($"No .csproj file referencing Microsoft.Quantum.Sdk found in workspace root {rootFolder}. " +
+                    "Will compile only .qs files in this folder.");
+                return defaultProject;
+            }
+
+            if (projects.Count() != 1)
```

Comment:
If we're actually supporting multiple projects. Why enable this when there is only one?

Reply:
Again it's a limitation of the way we treat `.csproj` files at the moment. We assume that any project compiles source files with scope `**\*.qs`, so two in the same folder would conflict.

But, this is a good point, I think this check should be moved _below_ the check for the auto-load property. (It's fine if multiple projects are in a folder, as long as we only auto-load one of them.)